doc/developer/design/20240508_subsource_schemas_after_planning.md

```diff
@@ -0,0 +1,113 @@
+# Restructure + genericize subsource planning
+
+-   Associated:
+    -   https://github.com/MaterializeInc/materialize/issues/20208
+    -   https://github.com/MaterializeInc/materialize/pull/26881
+    -   https://github.com/MaterializeInc/materialize/issues/26999
+
+## The Problems
+
+-   To complete #20208, we have to know the schema of the Kafka topic. This
+    schema is most readily knowable after planning the `SOURCE`.
+-   Subsource planning (generating `CreateSubsourceStatement`) currently happens
+    during purification before planning the `SOURCE`.
+-   Purification's current structure also means that we have to implement some
+    bits in purification (for new sources) and some in sequencing (when
```

Comment:
so the issue here is that we have to do similar things in both purification and in sequencing based on whether we are creating a new source vs adding a subsource/table to an existing source?

Reply:
Yes